app/main/user-agent-service.js

```diff
@@ -134,6 +144,21 @@ function configure(app: any, storage: ProfileStorage) {
   }));
 
   router.get('/visits', wrap(async function(req, res) {
+    req.checkQuery('limit').isInt().notEmpty();
+    const errors = req.validationErrors();
+    if (errors) {
+      res.status(401).json(errors);
+      return;
+    }
+
+    const limit = parseInt(req.query.limit, 10) || Number.MAX_SAFE_INTEGER;
+
+    // TODO: this should return full-fledged page objects.
+    const pages = await storage.visited(0, limit);
+    res.json({ pages });
+  }));
+
+  router.get('/awesome', wrap(async function(req, res) {
```

Comment:
I'm guessing this is for the awesomebar completion, but we may want completions elsewhere, maybe a better name?